firefoxThemes/DokiThemeDefinitions.js

```diff
@@ -1 +1 @@
"#363e3e","selectionForeground":"#fbfbfb","headerColor":"#363b3d","baseBackground":"#3b3f41","borderColor":"#424648","buttonColor":"#414648","selectionInactive":"#324142","identifierHighlight":"#324142","selectionBackground":"#295043","selectionBackgroundTransparent":"#295043AA","buttonFont":"#ada89c","foregroundColor":"#b0b0b0","highlightColor":"#324344","disabledColor":"#6f6f6f","accentColorTransparent":"#4ed4b05A","accentColorLessTransparent":"#4ed4b09A","accentColorMoreTransparent":"#4ed4b02A","accentColor":"#4ed4b0","accentContrastColor":"#1D4230","infoForeground":"#b0a17f","startColor":"#595959","stopColor":"#597cee","testScopeColor":"#2e3f3a","popupMask":"#928d8d05","codeBlock":"#323f41","foldedTextBackground":"#3c4443","textEditorBackground":"#35393b","classNameColor":"#89acf6","htmlTagColor":"#788a8b","stringColor":"#F6E9CB","keyColor":"#61ead9","keywordColor":"#9f73ea"},"overrides":{"theme":{"colors":{},"properties":{"ntp_background_alignment":"right"}}}},"5fb9c0a4-e613-457c-97a5-6204f9076cef":{"information":{"id":"5fb9c0a4-e613-457c-97a5-6204f9076cef","name":"Shima Rin","displayName":"Rin","dark":true,"author":"Unthrottled","group":"Yuru Camp","stickers":{"default":"shima_rin_dark.png"},"backgrounds":{"primary":"waifus/Shima_Rin/images/shima_rin_dark.png"},"jsonPath":"waifus/Shima_Rin/theme.json"},"colors":{"constantColor":"#ec9cb2","parameterColor":"#FFB86C","comments":"#687380","errorColor":"#ff5555","terminal.ansiRed":"#E356A7","terminal.ansiGreen":"#42E66C","terminal.ansiYellow":"#EFA554","terminal.ansiBlue":"#9B6BDF","terminal.ansiMagenta":"#E64747","terminal.ansiCyan":"#75D7EC","ghostColor":"#FFFFFF00","tagColor":"#7A7A7A","lineNumberColor":"#5a626c","foregroundColorEditor":"#F8F8F2","unusedColor":"#545c5b","diff.deleted":"#30383E","diff.conflict":"#3E2424","diff.inserted":"#223B28","diff.modified":"#253F4C","caretRow":"#313a49","baseIconColor":"#6c747d","contrastColor":"#404a57","nonProjectFileScopeColor":"#33393c","secondaryBackground":"#2c3039","selectionForeground":"#fbfbfb","headerColor":"#2e343e","baseBackground":"#323742","borderColor":"#393e48","buttonColor":"#3a4148","selectionInactive":"#403545","identifierHighlight":"#373c47","selectionBackground":"#4f3a4e","selectionBackgroundTransparent":"#4f3a4eAA","buttonFont":"#ada89c","foregroundColor":"#b0b0b0","highlightColor":"#3d4754","disabledColor":"#5b606f","accentColorTransparent":"#6688ad5A","accentColorLessTransparent":"#6688ad9A","accentColorMoreTransparent":"#6688ad2A","accentColor":"#6688ad","accentContrastColor":"#182f42","infoForeground":"#b0a17f","startColor":"#595959","stopColor":"#e66895","testScopeColor":"#2e3a3c","popupMask":"#928d8d05","codeBlock":"#3c3443","foldedTextBackground":"#323743","textEditorBackground":"#2b303b","editorAccentColor":"#8bd9ff","classNameColor":"#89acf6","htmlTagColor":"#6c7c96","stringColor":"#F6E9CB","keyColor":"#61ead9","keywordColor":"#ea73d6"},"overrides":{"theme":{"colors":{},"properties":{"ntp_background_alignment":"right"}}}}};
```

Comment:
Added a `export` keyword here in order to take advantage of JavaScripts modules. *And to `import` DokiThemeDefinitions into the `resource.js` background script.*

Reply:
Just as a heads up, I might remove this file from version control in the future.

Reply:
Will *DokiThemeDefinitions* be hosted on a server and get retrieved from there? If so, then a service worker system will have to be established. *This will probably fit nicely with the new [Firefox Manifest V3](https://blog.mozilla.org/addons/2021/05/27/manifest-v3-update/)*

> Background pages will be replaced by background service workers 

I'll say wait on adding service workers though until this is officially released.

Reply:
> Will DokiThemeDefinitions be hosted on a server and get retrieved from there?

It will still be bundled with the extension itself. Since it is an automatically generated file, there really is not a point for it to hang out in version control. I'll probably do that when I refactor the build script to be more comprehensible.